torch/linalg/__init__.py

```diff
@@ -140,6 +140,110 @@
+represented by a namedtuple (eigenvalues, eigenvectors).
+
+This function calculates all eigenvalues (and vectors) of :attr:`input`
+such that :math:`\text{input} = V \text{diag}(e) V^H`, where :math:`e` is eigenvalues tensor
+and :math:`V` is eigenvectors tensor.
+
+Since the input matrix :attr:`input` is supposed to be Hermitian,
+only the lower triangular portion is used by default
+and the imaginary part of the diagonal will always be treated as zero.
+
+.. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
+
+.. note:: The eigenvalues/eigenvectors are computed using LAPACK/MAGMA routines ``_syevd``, ``_heevd``.
+
+Args:
+    input (Tensor): the input tensor of size :math:`(*, n, n)` consisting of Hermitian
+                    :math:`n \times n` matrices, where `*` is zero or more batch dimensions.
+    UPLO ('L', 'U', optional): controls whether to consider upper-triangular or lower-triangular part.
+        Default: ``'L'``
+
+Returns:
+    (Tensor, Tensor): A namedtuple (eigenvalues, eigenvectors) containing
+
+        - **eigenvalues** (*Tensor*): Shape :math:`(*, m)`.
+            The eigenvalues in ascending order, each repeated according to its multiplicity.
+        - **eigenvectors** (*Tensor*): Shape :math:`(*, m, m)`.
+            The orthonormal eigenvectors of the ``input``.
+
+Examples::
+
+    >>> a = torch.randn(2, 2, dtype=torch.complex128)
+    >>> a = a + a.t().conj()  # creates a Hermitian matrix
+    >>> a
+    tensor([[2.9228+0.0000j, 0.2029-0.0862j],
+            [0.2029+0.0862j, 0.3464+0.0000j]], dtype=torch.complex128)
+    >>> w, v = torch.linalg.eigh(a)
+    >>> w
+    tensor([0.3277, 2.9415], dtype=torch.float64)
+    >>> v
+    tensor([[-0.0846+-0.0000j, -0.9964+0.0000j],
+            [ 0.9170+0.3898j, -0.0779-0.0331j]], dtype=torch.complex128)
+    >>> torch.allclose(torch.matmul(v, torch.matmul(w.to(v.dtype).diag_embed(), v.t().conj())), a)
+    True
+
+    >>> a = torch.randn(3, 2, 2, dtype=torch.float64)
+    >>> a = a + a.transpose(-2, -1)  # creates a symmetric matrix
+    >>> w, v = torch.linalg.eigh(a)
+    >>> torch.allclose(torch.matmul(v, torch.matmul(w.diag_embed(), v.transpose(-2, -1))), a)
+    True
+""")
+
+eigvalsh = _add_docstr(_linalg.linalg_eigvalsh, r"""
+linalg.eigvalsh(input, UPLO='L') -> Tensor
+
+This function returns eigenvalues of a complex Hermitian (conjugate symmetric)
+or real symmetric matrix :attr:`input`. The eigenvalues are returned in ascending order,
+each repeated according to its multiplicity.
+
+.. note:: The eigenvalues of real symmetric or complex Hermitian matrices are always real.
+
+.. note:: The eigenvalues are computed using LAPACK/MAGMA routines ``_syevd``, ``_heevd``.
+
+.. note:: This function doesn't support backpropagation, please use :func:`torch.linalg.eigh` instead.
+
+Args:
+    input (Tensor): the input tensor of size :math:`(*, n, n)` consisting of Hermitian
```

Comment:
Same notes here as above.

One option for the documentation of eigvalsh is to simply refer to the documentation for eigh and explain that this is non differentiable and that it only returns the eigen values. That might makes things simpler and ensure consistency.